shell/run_shell_cmd_test.go

```diff
@@ -53,3 +53,13 @@ func TestRunShellOutputToStderrAndStdout(t *testing.T) {
 	assert.True(t, strings.Contains(stderr.String(), "Terraform"), "Output directed to stderr")
 	assert.True(t, len(stdout.String()) == 0, "No output to stdout")
 }
+
+func BenchmarkPerformanceOfGitTopLevelDir(b *testing.B) {
```

Comment:
Hi, will be helpful to have a functional test that will validate cache usage

Reply:
Agree, I'll add it

Reply:
Okay, I have added a test 👍